libraries/helpers.rb

```diff
@@ -98,11 +98,7 @@ def get_1st_partition(device)
 # Get vpc-ipv4-cidr-blocks
 #
 def get_vpc_ipv4_cidr_blocks(eth0_mac)
-  token = get_metadata_token
-  uri = URI("http://169.254.169.254/latest/meta-data/network/interfaces/macs/#{eth0_mac.downcase}/vpc-ipv4-cidr-blocks")
-  vpc_ipv4_cidr_blocks = get_metadata_with_token(token, uri)
-  # Parse into array
-
+  vpc_ipv4_cidr_blocks = node['ec2']['network_interfaces_macs'][eth0_mac.downcase]['vpc_ipv4_cidr_blocks']
```

Comment:
how is node['ec2'] populated? we need to make sure it uses IMDSv2 and not v1.

From the [code]( https://github.com/chef/ohai/blob/main/lib/ohai/mixin/ec2_metadata.rb) it seems to support IMDSv2 but we need to check what version of ohai is used by our chef-client version

Reply:
https://github.com/chef/ohai/blob/main/CHANGELOG.md -- from here it looks like they switched to using `imdsv2` in v17.0.42 ... we're using Chef version 17.2.29 -- but I'm not sure how to determine which version of ohai we're using (or if the numbers are even related?)